src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private int size;
+    private int threshold;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> current = new Node<>(key, value, null);
+        Node<K, V> nodeInTable = table[index];
+        if (table[index] == null) {
+            table[index] = current;
+        } else {
+            while (nodeInTable.next != null || Objects.equals(nodeInTable.key, key)) {
+                if (Objects.equals(nodeInTable.key, key)) {
+                    nodeInTable.value = value;
+                    return;
+                }
+                nodeInTable = nodeInTable.next;
+            }
+            nodeInTable.next = current;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+        if (current == null) {
+            return null;
+        }
+        while (current.next != null || Objects.equals(current.key, key)) {
```

Comment:
```suggestion
        while (current != null) {
```
let's try to remove the `if` above